### ISSUE TITLE
Hook "session-closed" not triggered for with the "-t" option for a target pane on Fedora 40.

### ISSUE THREAD

**Issue by twoGiants** [tmux/tmux#4267]
### Issue description

Running a bash script or shell command on a `session-closed` hook for a target pane is not working. It works only if I use the `-g` option and set the hook globally.

I will provide the steps to reproduce without a config:

1. Start tmux without a config and with logs:

```bash
tmux -vv -Ltest -f/dev/null new
```

2. Set the hook:

```bash
tmux set-hook -t 0 session-closed 'run-shell "echo success >> ~/tmux-session-close-log.txt"'
```

3. Kill the session:

```bash
tmux kill-sess -t 0
```

4. Check if the file is there:

```bash
cat ~/tmux-session-close-log.txt
```

You should get:

```bash
cat: /home/<your-user-name>/tmux-session-close-log.txt: No such file or directory
```

If you do the same but set the hook globally with:

```bash
tmux set-hook -g session-closed 'run-shell "echo success >> ~/tmux-session-close-log.txt"'
```

You should get:

```bash
cat tmux-session-close-log.txt 
success
```

### Required information

Please provide the following information:

* tmux version (`tmux -V`).
  * ==> `tmux 3.5a`
* Platform (`uname -sp`).
  * ==>  `Linux unknown`
  * ==> I am running _Fedora 40 Workstation Edition_.
* $TERM inside and outside of tmux (`echo $TERM`).
  * ==> inside: `tmux-256color` , outside: `xterm-256color` 
* Logs from tmux (`tmux kill-server; tmux -vv new`).

[tmux-out-10375.log](https://github.com/user-attachments/files/17879279/tmux-out-10375.log)
[tmux-server-10375.log](https://github.com/user-attachments/files/17879280/tmux-server-10375.log)
[tmux-client-10373.log](https://github.com/user-attachments/files/17879281/tmux-client-10373.log)

**Comment by twoGiants** [comment]
Can you support @nicm? Thank you!

**Comment by twoGiants** [comment]
Thanks for the workaround! I am actually already using something similar, but with session name instead of id which would be a bit more stable but its good enough in my case:

`tmux set-hook -g session-closed 'run-shell "if [ \"#{hook_session_name}\" = \"<my-session-name>\" ]; then ~/./shutdown.sh; fi"'`

**Comment by nicm** [comment]
Let me know if it fixes it but it causes tmux to crash on shutdown so I'll need to look at that...

**Comment by twoGiants** [comment]
@nicm Thank you! Because I am new here I have a question about your patch. You give me the code so I guess I am supposed clone the repo, apply the patch and install [from version control](https://github.com/tmux/tmux?tab=readme-ov-file#from-version-control)? And if all works well I let you know and you add that to the latest release?

**Comment by nicm** [comment]
Yes check out master then you apply the patch and build. You do not need to install you can run from the source directory with "./tmux".

**Comment by twoGiants** [comment]
Perfect, thanks. Will do and let you know.
